src/harness/testcases/WINNF_FT_S_GRA_testcase.py

```diff
@@ -58,6 +58,15 @@ def setUp(self):
   def tearDown(self):
     pass
 
+  @classmethod
+  def setUpClass(self):
```

Comment:
Same comments as above, plus you won't need this for the majority of the GRA test cases so it should probably be part of GRA.5,6 individually rather than applying to all GRA test cases.

Reply:
Already rewrite and I finished startserver and stopserver function and put them into the util.py file. So in the test cases, if the test case need to startserver, it can simply do it by giving the url input parameter.